compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -3059,6 +3059,12 @@ private BRecordType createErrorDetailRecordType(BLangInvocation iExpr, boolean r
     }
 
     private void checkObjectFunctionInvocationExpr(BLangInvocation iExpr, BObjectType objectType) {
+        if (objectType.getKind() == TypeKind.SERVICE &&
+                !(iExpr.expr.getKind() == NodeKind.SIMPLE_VARIABLE_REF &&
+                (Names.SELF.equals(((BLangSimpleVarRef) iExpr.expr).symbol.name)))) {
+            dlog.error(iExpr.pos, DiagnosticCode.SERVICE_FUNCTION_INVALID_INVOCATION);
+            return;
```

Comment:
Do we have to return here? Can we integrate this with the object init check https://github.com/ballerina-platform/ballerina-lang/pull/18616/files#diff-1f0acc81bd72ee2f8368b0a25cd59e9cR3085

Reply:
As we discussed offline, this is not possible due to services using buit-in type. Created an issue to track this.
https://github.com/ballerina-platform/ballerina-lang/issues/18717